### PR TITLE
IcsToast. Use IcsToast for tooltip on long press on action menu items

### PR DESCRIPTION
**Pull request by Prototik**

### CODE REVIEW

File: actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        TextView view = new TextView(context);
+        view.setText(s);
+        view.setTextColor(0xFFDADADA);
+        view.setGravity(Gravity.CENTER);
+        view.setBackgroundResource(R.drawable.abs__toast_frame);
+        toast.setView(view);
+        return toast;
+    }
+
+    public static IcsToast makeText(Context context, int resId, int duration) {
```
Comment (JakeWharton) [inline actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java]: This version doesn't seem to be used. It is provided for others to use (despite being in the `internal` package)?

Comment (Prototik) [inline actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java]: Hm. Yes, don't used, but working. Can be used by devs.
I can move IcsToast to a com.actionbarsherlock.widget package. Is this necessary?

Comment (JakeWharton) [inline actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java]: No. It should definitely be kept internal.

**Comment by JakeWharton** [comment]
The Travis build is failing but it's not related to your pull. I'll verify manually tomorrow morning.

File: actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java (hunk 1 above)
Comment (Prototik) [inline actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java]: You suggest remove the method makeText(Context, int, int)?

**Comment by Prototik** [comment]
What about this pull request?

File: actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+import com.actionbarsherlock.R;
+
+public class IcsToast extends Toast {
+    public static final int LENGTH_LONG = Toast.LENGTH_LONG;
+    public static final int LENGTH_SHORT = Toast.LENGTH_SHORT;
+    private static final String TAG = "Toast";
+
+    public static Toast makeText(Context context, CharSequence s, int duration) {
+        if(VERSION.SDK_INT >= VERSION_CODES.HONEYCOMB) {
```
Comment (JakeWharton) [inline actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java]: Space after `if`. Also the name of the class is `IcsToast` but we're only using on pre-Honeycomb. Should probably switch the conditional to compare against ICS.

File: actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        return makeText(context, context.getResources().getString(resId), duration);
+    }
+
+    public IcsToast(Context context) {
+        super(context);
+    }
+
+    @Override
+    public void setText(CharSequence s) {
+        if(VERSION.SDK_INT >= VERSION_CODES.HONEYCOMB) {
```
Comment (JakeWharton) [inline actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java]: Space after `if`. Same HC/ICS stuff as above.

File: actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            super.setText(s);
+            return;
+        }
+        if (getView() == null) {
+            return;
+        }
+        try {
+            ((TextView) getView()).setText(s);
+        } catch (ClassCastException e) {
+            Log.e(IcsToast.TAG, "This Toast was not created with IcsToast.makeText", e);
```
Comment (JakeWharton) [inline actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java]: Drop `IcsToast.` so it's just `TAG`.

File: actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    public static Toast makeText(Context context, CharSequence s, int duration) {
+        if(VERSION.SDK_INT >= VERSION_CODES.HONEYCOMB) {
+            return Toast.makeText(context, s, duration);
+        }
+        IcsToast toast = new IcsToast(context);
+        toast.setDuration(duration);
+        TextView view = new TextView(context);
+        view.setText(s);
+        view.setTextColor(0xFFDADADA);
```
Comment (JakeWharton) [inline actionbarsherlock/src/com/actionbarsherlock/internal/widget/IcsToast.java]: Out of curiosity where did you get this value from?

**Comment by Prototik** [comment]
@JakeWharton fixed.

**Pull request merged**